ga4gh/cli.py

```diff
@@ -1496,6 +1499,29 @@ def configtest_main(parser=None):
 ##############################################################################
 
 
+def getNameFromPath(filePath):
+    """
+    Returns the filename of the specified path without its extensions.
+    This is usually how we derive the default name for a given object.
+    """
+    if len(filePath) == 0:
+        raise ValueError("Cannot have empty path for name")
+    fileName = os.path.split(filePath)[1]
+    # We need to handle things like .fa.gz, so we can't use
+    # os.path.splitext
+    ret = fileName.split(".")[0]
```

Comment:
Except that this is exactly what causes server failure on fairly common cases where files have `.`s in their name. I'd propose instead something like:
```
def getNameFromPath(filePath, extensions=[]):
```
where `extensions` would be an array of possible extensions to expect, ex: `['.fa','.fa.gz']` and the function would _only_ strip out expected extensions, leaving any additional `.`s in the name intact.